.github/workflows/build_and_run_test.yml

```diff
@@ -37,23 +33,41 @@ jobs:
         tmpdir=$(mktemp -d /tmp/tmp.XXXXXX)
         echo "::set-output name=tmpdir::$tmpdir"
 
+    - name: Start docker containers
+      run: |
+        cd docker/generated
+        docker-compose up -d --build
+
     - name: Run integration test
       run: |
         tmpdir=${{ steps.set-tmpdir.outputs.tmpdir }}
         ts-tests/scripts/run-test.sh $tmpdir
       shell: bash {0}
 
-    - uses: actions/upload-artifact@v2
+    - name: Archive logs if test fails
+      uses: actions/upload-artifact@v2
       if: ${{ failure() }}
       with:
         name: litentry-parachain-debug.log
         path: ${{ steps.set-tmpdir.outputs.tmpdir }}/**/*
         retention-days: 7
 
+    - name: Push docker image if test passes
+      if: ${{ success() }}
+      run: |
+        [ -z "$(docker images --filter=dangling=true -q)" ] || docker rmi -f $(docker images --filter=dangling=true -q)
+        docker images
+        docker push -a litentry/litentry-parachain
+
     - name: Clean up
       if: ${{ always() }}
       run: |
         tmpdir=${{ steps.set-tmpdir.outputs.tmpdir }}
-        ts-tests/scripts/clean-up.sh $tmpdir
+        docker images
+        cd docker/generated
+        docker-compose rm -f -s -v
+        docker volume ls | grep generated_ | sed 's/local *//' | xargs docker volume rm
+        [ -z "$(docker images --filter=dangling=true -q)" ] || docker rmi -f $(docker images --filter=dangling=true -q)
+        docker rmi -f $(docker images litentry/litentry-parachain --format "{{.Repository}}:{{.Tag}}" | grep -v latest)
```

Comment:
Doesn't have to remove image since we can reuse them in this future building. It can save time since we don't have to push the same image everytime

Reply:
> Doesn't have to remove image since we can reuse them in this future building. It can save time since we don't have to push the same image everytime

it actually keeps litentry/litentry-parachain:latest for further usage (while removing other tagged version)